src/express-server/constants/ErrorMessages.ts

```diff
@@ -3,4 +3,13 @@ export enum ErrorMessages {
   "NOPRODUCTS"= "There is no products to show",
   "NOREVIEWS" = "There is no reviews for selected products",
   "NOUSERS" = "There is no users to show",
+  "NOTFOUND" = "Not found",
+  "NOUSER" = "User with such email isn't register",
+  "INCORRECTPASSWORD" = "Password isn't correct",
+  "NOTOKEN" = "There is no token",
+  "NOTVERIFIEDTOKEN" = "The token is not verified",
+  "FAILEDAUTH" = "Authorization is failed",
+  "FAILDEDGOOGLEAUTH" = "Google+ user is not authorized",
+  "FAILEDLOCALAUTH" = "User is not authorized",
+  "WRONGCREDENTIALS" = "Wrong credentials",
```

Comment:
Let's do next - create your custom AuthError class like
`class AuthError extends Error {}` and implement some methods basing on logic like e.g. `AuthError.failedAuth(resource)`, where message will be one of `Local/Google/Facebook` and should return the error with message like `Local auth is failed`. Separating Error logic is good for stacktracing in future to better understand the source of an error not only by message

`throw new AuthError(NOUSER)`

Reply:
This is how it's implemented on my project
```
export class ValidationError extends Error {
  public static fromSingleParam(paramName: string, message: string) {
    return new ValidationError(`Error in parameter ${paramName}: ${message}`, {
      [paramName]: [message],
    });
  }

  constructor(
    public message: string = 'ValidationError',
    public details: object = null,
  ) {
    super(message);
  }
}
```